File: Gymify.Services/Database/UserReward.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Gymify.Services.Database
{
    public class UserReward
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey(nameof(UserId))]
        public int UserId { get; set; }

        public User? User { get; set; }

        [ForeignKey(nameof(RewardId))]
        public int RewardId { get; set; }

        public Reward? Reward { get; set; }

        public String? Code {get; set;}

        public DateTime RedeemedAt { get; set; } = DateTime.UtcNow;

        public bool IsUsed { get; set; } = false;
    }
}

File: UI/gymify_desktop/lib/models/user_reward.dart
import 'package:gymify_desktop/models/reward.dart';
import 'package:gymify_desktop/models/user.dart';
import 'package:json_annotation/json_annotation.dart';


part 'user_reward.g.dart';

@JsonSerializable()
class UserReward {
  final int id;

  final int userId;
  final User? user;

  final int rewardId;
  final Reward? reward;

  final String? code;

  final DateTime redeemedAt;

  final String status;

  UserReward({
    required this.id,
    required this.userId,
    this.user,
    required this.rewardId,
    this.reward,
    this.code,
    required this.redeemedAt,
    required this.status,
  });

  bool get isUsed => status == "Used";

  bool get isExpired => status == "Expired";

  bool get isActive => status == "Active";

  factory UserReward.fromJson(Map<String, dynamic> json) =>
      _$UserRewardFromJson(json);

  Map<String, dynamic> toJson() => _$UserRewardToJson(this);
}